### ISSUE TITLE
Some achievements can cause a crash, garbage or no achievements to show

### ISSUE THREAD

**Issue by fr500** [libretro/RetroArch#4860]
## Description

Both megaman X 1.1 and the Rockman & Forte translation seem to have this issue, you'll either get no achievements (empty list), garbage, or a crash when browsing the list.

![retroarch_20170426_215417](https://cloud.githubusercontent.com/assets/1721040/25550700/e2cb6b7c-2c40-11e7-80b1-9c6d5a1ae641.png)

Seems completely fine with most games that are not these two...

### Expected behavior

Don't crash, show the list properly

### Steps to reproduce the bug

1. Load RetroArch
2. Load Megaman X 1.1 with Snes9x
3. Load achievement list

Repeat a few times to get the crash and/or corrupgion

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit]

### Environment information

- OS: [The operating system you're running]
- Compiler: [In case you are running local builds]

**Comment by fr500** [comment]
@leiradel ^^

**Comment by leiradel** [comment]
@fr500 I can't seem to find the correct ROM:

```
RetroArch [INFO] :: CHEEVOS testing SNES (8Mb padding).
RetroArch [INFO] :: CHEEVOS getting game id for hash 107ec289a82bdd6be00ad5f627f116d3
RetroArch [INFO] :: CHEEVOS url to get the game's id: http://retroachievements.org/dorequest.php?r=gameid&m=107ec289a82bdd6be00ad5f627f116d3
RetroArch [INFO] :: CHEEVOS got game id 0.
RetroArch [INFO] :: CHEEVOS testing NES (discards VROM).
RetroArch [INFO] :: CHEEVOS testing Generic (plain content).
RetroArch [INFO] :: CHEEVOS getting game id for hash 8998f2d66a84c1d44ab6387db7314efd
RetroArch [INFO] :: CHEEVOS url to get the game's id: http://retroachievements.org/dorequest.php?r=gameid&m=8998f2d66a84c1d44ab6387db7314efd
RetroArch [INFO] :: CHEEVOS got game id 0.
RetroArch [INFO] :: CHEEVOS this game doesn't feature achievements.
```

Mine has 1,573,376 bytes, 512 bytes more than exactly 1.5 MB, which strikes me as odd...

**Comment by fr500** [comment]
@leiradel now I tried RAsnes
RAsnes gives me this checksum:
7f966d69bbf7f1e639bf37afe9ce275f

RetroArch gives me two different checksums for some reason, one after another:
RetroArch [INFO] :: CHEEVOS getting game id for hash 9a296c29afcae496c61a615729758f4b
RetroArch [INFO] :: CHEEVOS getting game id for hash df1cc0c8c8c4b61e3b834cc03366611c

**Comment by leiradel** [comment]
Yes, that RAsnes hash works. It's odd. I'll check in RAsnes' source code if the hash method has changed.

**Comment by fr500** [comment]
Hmmm not sure if this is still happening asssigning myself so I can keep track

**Comment by meleu** [comment]
Unable to reproduce here.
